server/helpers/commonHelper.js

```diff
@@ -0,0 +1,120 @@
+const split = require('lodash/split');
+const keys = require('lodash/keys');
+const groupBy = require('lodash/groupBy');
+const get = require('lodash/get');
+const concat = require('lodash/concat');
+
+const { getProjects } = require('./testrailReportHelper');
+
+const { parseTestCase, extractTestCase } = require('./parserHelper');
+const { createSection, getProjectRepoSections } = require('./sectionHelper');
+const { getProjectRepoCases, createSuiteCases } = require('./testCaseHelper');
+const logger = require('../utils/logger');
+const {
+  createTestSuite,
+  getSuiteName,
+  getProjectSuites,
+} = require('./testSuiteHelper');
+
+const getParsedTestCase = data => {
+  const { content, testName } = data;
+  const testCaseBlock = extractTestCase(content, 'test_case');
+
+  if (!testCaseBlock) {
+    logger.log('warn', `TEST CASE FOR IT BLOCK '${testName}' NOT FOUND`);
+    return null;
+  }
+
+  return parseTestCase(testCaseBlock, testName);
+};
+
+const getRepoProjectSuiteNames = data =>
+  concat(
+    map(get(data, 'repoProjectSuites', []), repoProjectSuite =>
+      getSuiteName(repoProjectSuite),
+    ),
+    get(data, 'repoProjectCommonSuiteNames', []),
+  );
+
+// in the future, we will detect not needed suites and remove them
+const removeAndCreateTestSuiteSections = data => {
+  const { describes, projectName, repoProjectCommonSuiteNames } = data;
+  const repoProjectSuiteNames = getRepoProjectSuiteNames({
+    repoProjectSuites: getProjectSuites({ suites: describes, projectName }),
+    repoProjectCommonSuiteNames,
+  });
+
+  const suiteSectionPromises = map(
+    repoProjectSuiteNames,
+    createRepoProjectSuiteName =>
+      createSuiteAndSection({
+        testName: createRepoProjectSuiteName,
+        projectName,
+      }),
+  );
+
+  return Promise.all(suiteSectionPromises);
+};
+
+const createSuiteAndSection = async data => {
+  const { testName, projectName } = data;
+
+  const suite = await createTestSuite({
+    name: testName,
+    projectName,
+  });
+
+  const section = await createSection({
+    name: testName,
+    suite,
+    projectName,
+  });
+
+  return {
+    section,
+    suite,
+  };
+};
+
+const createProjectCases = async data => {
+  const { createProjectRepoCases, projectName } = data;
+  const groupedCreateProjectRepoSuiteCases = groupBy(
+    createProjectRepoCases,
+    'suite_id',
+  );
+
+  let sections = await getProjectRepoSections({
+    projectName,
+    suiteIds: keys(groupedCreateProjectRepoSuiteCases),
+  });
+  sections = groupBy(sections, 'suite_id');
+
+  const promises = map(
+    groupedCreateProjectRepoSuiteCases,
+    (createProjectRepoSuiteCases, suiteId) =>
+      createSuiteCases({
+        createProjectRepoSuiteCases,
+        projectName,
+        section: head(sections[suiteId]),
+      }),
+  );
+  return Promise.all(promises);
+};
+
+// in the future, we will prepare test cases for removing, updating and creating
+const getPreparedProjectRepoCases = data => {
```

Comment:
change naming 
repo => project => entity (in code)
project => entity (in testrails)